src/core/public/apm_system.ts

```diff
@@ -39,10 +40,10 @@ export class ApmSystem {
   private readonly enabled: boolean;
   /**
    * `apmConfig` would be populated with relevant APM RUM agent
-   * configuration if server is started with `ELASTIC_APM_ACTIVE=true`
+   * configuration if server is started with elastic.apm.* config.
    */
   constructor(private readonly apmConfig?: ApmConfig) {
-    this.enabled = process.env.IS_KIBANA_DISTRIBUTABLE !== 'true' && apmConfig != null;
+    this.enabled = apmConfig != null && !!apmConfig.active;
```

Comment:
I had a hard time debugging the config. It's turned out we don't read values from `kibana.dev.yml`. It might be another source of confusion for developers.

Reply:
> It's turned out we don't read values from `kibana.dev.yml`

I think there was some confusion on whether or not we still supported this. It appears we do. Can fix as a follow up so I can get this in for FF

Reply:
Opened issue: https://github.com/elastic/kibana/issues/79490
Should be a pretty quick follow up